lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
I like event schema definition but something about event schema validations doesn't feel quite right. Another option could be to trigger validation on `Event.new`, and have some other way to load historical events (not sure if this is better just another thought). I wrote up some thoughts about how I'd like to change the modelling of Event in this issue - https://github.com/envato/event_sourcery/issues/139